101.md

```diff
@@ -0,0 +1,103 @@
+```js
+{
+    "kind": 30168,
+    "content" : ""
+    "tags": [
+      ["d", "<form identifier>"],
+      ["name", "Name of the form"],
+      ["settings", JSON.stringify({description: "<description of the form."})],
+      ["field", "<fieldId>","<input-type>","<label for the field>","<Options (for option type)>", "<stringified JSON settings>"],
+      ["field", "<fieldId>", "option", "label for options field", 
+        JSON.stringify([["<optionId1>", "option label", "<optionId2>", "option label"]]),
+        "<stringified JSON settings>"
+      ]
+    ],
+    "pubkey": "<Author of the form>"
+}
+```
+
+The different tags used to describe the form are described as:
+
+|Tags|Description|
+|----|--------------
+|d|The unique identifier of a form, for a user|
+|name| A text serving as the name of the form|
+|settings| An optional global config for the form.
+|field| Contains the following values<table><tr><th>index</th><th>name</th><th>description</th></tr><tbody><tr><td>1</td><td>FieldId</td><td>an id that uniqely identifies a field in the forn</td></tr><tr><td>2</td><td>input-type</td><td>A field that describes the type of value that can be expected as a response to this field, values can be: `text`, `option` or `label`</td></tr><tr><td>3</td><td>label</td><td>A label for the field</td></tr><tr><td>4</td><td>options</td> <td>Only used for input-type option, is a Json stringified array of strings containing [`<OptionId`>, <`label`>, `<optional config>`] </td> </tr><tr><td>5</td><td>fieldSettings</td><td>An optional JSON stringified object that contains settings specific to the field, for example `renderElement`, a setting which indicates what UI element to render to the client </td></tr>
+</tbody></table> 
+
+
+## Responses - Public
+
+Response events are attached to a form(3068 kind event), and the event data is stored in a `kind: 30169` event
+
+Response structure:
+
+```js
+{
+    "kind": 30169,
+    "content" : "",
+    "tags": [["a", "30168:<pubkey of the author>:<form identifier>"],
+    ["response", "<FieldId>", "<response as string>", "<stringified metadata object>"]],
+    "pubkey": "Author of Response"
+}
+```
+
+for option fields, the response is the id of the option selected. In case of multiple-choice fields these id's maybe delimited by a semi-colon, ";", For example:
+
+```js
+{
+    "kind": 30169,
+    "content" : "",
+    "tags": [["a", "30168:<pubkey of the author>:<form identifier>"],
+    ["response", "<FieldId>", "ZCZC;XCZXCZ;Z34Z", "<stringified metadata object>"]],
+    "pubkey": "Response author"
+}
+```
+
+## Access Control
+
+In general access is granted by signing the event and encrypting the content with different keys. 
+the mechanism to share the keys is by encrypting the keys and adding it to the p-tag of the event.
+The p-tag for `kind:30168` events, should look like. 
+
+`["p","<pubkey for the user>", "<optional relays>", "<Encrypted-View-Key>", "<Encrypted-Signing-key>"]`
```

Comment:
Interesting choice of putting the two keys in every tag. It does reveal to the public who can edit and who can view. But maybe that's not a problem. Is the goal related to performance? To not decrypt all the other tags as in #1228 at the expense of a bigger event?

Reply:
yes in part, it also made syntactical sense while writing the client code.

Reply:
It does make simpler. We need to reflect on @mikedilger's concern on #1228 about not reusing the `p` tag for any of this.

Reply:
I've thought about it as well, doesn't seem good to overload the `p-tag`. I thought of an agreed upon `p<kindno>` tags to solve this, but not using p-tags also risks users not being notified, since how many kinds do clients look for to notify the user?

Reply:
@vitorpamplona @mikedilger NIP-72 also already overloads the `p-tag` https://github.com/coracle-social/nips/blob/groups/72.md?plain=1#L27

Reply:
NIP 72 is not a great spec, it should have used `moderator` or something rather than p tags

Reply:
It's from the times everyone was moving to markers and away from positional tags. Now everyone is moving away from markers and into custom tags, like the `q` thing. :)